vars/provisionNodes.groovy

```diff
@@ -0,0 +1,123 @@
+      nodeString = new_list.join(',')
+    } else if (config['node_count'] > node_cnt) {
+      echo "${config['node_count']} clients requested."
+      error "Only ${node_cnt} clients available!"
+    }
+  }
+
+  checkoutScm url: 'ssh://review.hpdd.intel.com:29418/exascale/jenkins',
+                                  checkoutDir: 'jenkins',
+                                  credentialsId: 'daos-gerrit-read'
+  def options = ''
+  def snapshot = ''
+  def wait_for_it = true
+  if (config['snapshot']) {
+    options += ' --snapshot'
+    wait_for_it = false
+  }
+  if (config['arch']) {
+    options += " --arch=${config['arch']}"
+  }
+  if (config['distro']) {
+    options += " --distro=${config['distro']}"
+  }
+  if (config['profile']) {
+    options += " --profile=${config['profile']}"
+  }
+
+  sshagent (credentials: ['daos-provisioner']) {
+
+    if (config['power_only']) {
+      sh script: """./jenkins/test_manager/node_powercycle.py \
+                     --node=${nodeString}""",
+         returnStatus: true
+    } else {
+      sh script: """./jenkins/test_manager/node_provision_start.py \
+                     --nodes=${nodeString} ${options}""",
+         returnStatus: true
+    }
+
+    def woptions = ''
+    if (wait_for_it) {
+      if (config['timeout']) {
+        if (config['timeout'] <= 0) {
+          wait_for_it = false
+        } else {
+          woptions = " --timeout=${config['timeout']}"
+        }
+      }
+      if (config['poll']) {
+        woptions = " --poll=${config['poll']}"
+      }
+    }
+    if (wait_for_it) {
+      def rc = 0
+      rc = sh script: """./jenkins/test_manager/wait_for_node_ready.py \
+                       --nodes=${nodeString} ${woptions}""",
+              returnStatus: true
+      if (rc != 0) {
+        error "One or more nodes failed to provision!"
+      }
+    }
+    // Prepare the node for daos/cart testing
+    def provision_script = '''set -ex
+                              groupadd -g 1101 jenkins
+                              useradd -b /localhome -g 1101 -u 1101 jenkins
+                              mkdir -p /localhome/jenkins/.ssh
+                              cat /tmp/id_rsa.pub >> /localhome/jenkins/.ssh/authorized_keys
+                              mv /tmp/id_rsa /localhome/jenkins/.ssh/
+                              chmod 700 /localhome/jenkins/.ssh
+                              chmod 600 /localhome/jenkins/.ssh/{authorized_keys,id_rsa}
+                              chown -R jenkins.jenkins /localhome/jenkins/.ssh
+                              echo \\"jenkins ALL=(ALL) NOPASSWD: ALL\\" > /etc/sudoers.d/jenkins
+                              yum copr -y enable jhli/ipmctl
+                              yum copr -y enable jhli/safeclib
+                              yum install -y libipmctl-devel openmpi CUnit fuse python34-PyYAML python34-nose python34-pip valgrind python34-paramiko'''
+    sh script: 'set -x; pdcp -R ssh -l root -w ' + config['NODELIST'] +
```

Comment:
This needs to be reworked.
SSH private keys access needs to be from ssh-agent(credentials: credential_id) only.
The daos-provisioner ID can be used for privileged access to the CI nodes.

Reply:
I changed that to [generate a session key](https://github.com/daos-stack/pipeline-lib/pull/13/commits/313151f2769576236c676cc0407956eb4db9f4f7).